core/modules/node/node.pages.inc

```diff
@@ -343,11 +341,22 @@ function node_form($form, &$form_state, Node $node) {
     '#submit' => array('node_form_submit'),
   );
   if (!empty($node->nid) && node_access('delete', $node)) {
+    $destination = array();
+    if (isset($_GET['destination'])) {
+      $destination = backdrop_get_destination();
+      unset($_GET['destination']);
+    }
+
+    // Use a link for the delete button so the form doesn't need to validate.
     $form['actions']['delete'] = array(
-      '#type' => 'submit',
-      '#value' => t('Delete'),
+      '#type' => 'link',
+      '#title' => t('Delete'),
+      '#href' => "node/{$node->nid}/delete",
+      '#options' => array(
+        'query' => $destination,
```

Comment:
The easiest implementation here would probably just be to change this line to:

```
'query' => isset($_GET['destination']) ? backdrop_get_destination() : array(),
```

Reply:
addressed in https://github.com/backdrop/backdrop/pull/2043